src/runtime/program/vote/execute.zig

```diff
@@ -428,6 +429,55 @@ fn executeAuthorizeChecked(
     );
 }
 
+/// Agave https://github.com/anza-xyz/agave/blob/24e62248d7a91c090790e7b812e23321fa1f53b1/programs/vote/src/vote_processor.rs#L114-L118
+fn executeUpdateValidatorIdentity(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+) (error{OutOfMemory} || InstructionError)!void {
+    try ic.info.checkNumberOfAccounts(2);
+
+    var new_identity = try ic.borrowInstructionAccount(
+        @intFromEnum(vote_instruction.UpdateVoteIdentity.AccountIndex.new_identity),
+    );
+    defer new_identity.release();
+
+    try updateValidatorIdentity(
+        allocator,
+        ic,
+        vote_account,
+        new_identity.pubkey,
+    );
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/24e62248d7a91c090790e7b812e23321fa1f53b1/programs/vote/src/vote_state/mod.rs#L722
+///
+/// Update the node_pubkey, requires signature of the authorized voter
+fn updateValidatorIdentity(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    new_identity: Pubkey,
+) (error{OutOfMemory} || InstructionError)!void {
+    const versioned_state = try vote_account.deserializeFromAccountData(
+        allocator,
+        VoteStateVersions,
+    );
+
+    var vote_state = try versioned_state.convertToCurrent(allocator);
+    defer vote_state.deinit();
+
+    // Both the current authorized withdrawer and new identity must sign.
+    if (!(ic.info.isPubkeySigner(vote_state.authorized_withdrawer) and
+        ic.info.isPubkeySigner(new_identity)))
+    {
+        return InstructionError.MissingRequiredSignature;
+    }
```

Comment:
small nit : feels like we can make this much easier to read like the following:

```suggestion        
        const withdrawer_signed = ic.info.isPubkeySigner(vote_state.authorized_withdrawer)
        const new_identity_signed = ic.info.isPubkeySigner(new_identity)
        // Both the current authorized withdrawer and new identity must sign.
        const has_required_sigs = withdrawer_signed and new_identity_signed
        if (!has_required_sigs) {
            return InstructionError.MissingRequiredSignature;
        }
```

also wdyt about renaming the field on vote_state from `authorized_withdrawer` to just `withdrawer`?

Reply:
Applied the variable binding suggestions [4b1dedc](https://github.com/Syndica/sig/pull/611/commits/4b1dedc7e541f361cfad83e46a4f6316d6aa9dd9)

> also wdyt about renaming the field on vote_state from authorized_withdrawer to just withdrawer?

I think that makes sense. The ` authorized_` is definitely redundant. But I think it would be cleaner to rename in a separate PR. There is also an `authorized_voters` field too that can be renamed to just `voters`. Added a TODO for this.